scipy/_lib/_util.py

```diff
@@ -141,9 +141,13 @@ def _lazywhere(cond, arrays, f, fillvalue=None, f2=None):
     temp1 = xp.asarray(f(*(arr[cond] for arr in arrays)))
 
     if f2 is None:
-        fillvalue = xp.asarray(fillvalue)
-        dtype = xp.result_type(temp1.dtype, fillvalue.dtype)
-        out = xp.full(cond.shape, fill_value=fillvalue, dtype=dtype)
+        # If `fillvalue` is a Python scalar and we convert to `xp.asarray`, it gets the
+        # default `int` or `float` type of `xp`, so `result_type` could be wrong.
+        # `result_type` should/will handle mixed array/Python scalars
+        # (data-apis/array-api#805) but doesn't yet. So in the meantime, this fails
+        # for array-api-strict.
+        dtype = xp.result_type(temp1.dtype, fillvalue)
```

Comment:
@j-bowhay I am at a loss for why this did not cause CI failure in this PR, but it is causing a failure now in gh-20971.  Any guesses?